calloptions.go

```diff
@@ -56,9 +56,10 @@ type CallOptions struct {
 	// to an instance of the intended service.
 	RoutingDelegate string
 
-	// callerName can only be used when forwarding a request. It can only be set internally,
-	// e.g. by calling (*InboundCall).CallOptions() when forwarding a request
-	callerName string
+	// CallerName is the name of the service issuing the call. It can also be used
```

Comment:
Should we mention that this for outbound calls, it's automatically set to the channel's service name, and should only be specified when that needs to be overridden (e.g., transparent proxying bit you have).